### PR TITLE
added static prop to remove event listener

### PR DESCRIPTION
**Pull request by AnnMarieW**
As reported on the forum,  components with event listeners can cause issues with screen readers.

https://community.plotly.com/t/how-to-remove-events-from-dash-html-components-to-avoid-misinterpretation-by-screen-readers/70477

This PR adds a `disable_n_clicks` prop to all html components, giving the ability to remove the event listener.  

By default, `n_clicks` will be enabled on any html component that has an `id`.  If `n_clicks` is not used in a callback for that component, the event listener can be removed by setting `disable_n_clicks`=True`  

For example:

- No event listener:
`html.Div()`
`html.Div(id="my-id", disable_n_clicks=True)`

- onClick event listener:
`html.Div(id="my-id")`

## Contributor Checklist

- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### optionals

- [x] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follows
    -  [ ] this GitHub [#PR number]() updates the dash docs
    -  [ ] here is the show and tell thread in Plotly Dash community

### CODE REVIEW

**Comment by AnnMarieW** [comment]
Needs a better name for the prop than `static` since that may imply that no callbacks can be done on the component.  Perhaps it should be `disable_n_clicks` ?

**Comment by T4rk1n** [comment]
> Needs a better name for the prop than `static` since that may imply that no callbacks can be done on the component. Perhaps it should be `disable_n_clicks` ?

I like `disable_n_clicks`,  it's more explicit than `static`.

On another note, I don't think `n_clicks` changes requires a re-render, perhaps add memo with a custom handler or refactor the component back to classes and implement `shouldComponentUpdate`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,11 +2,11 @@
 All notable changes to `dash` will be documented in this file.
 This project adheres to [Semantic Versioning](https://semver.org/).
 
-## Unreleased
+## [UNRELEASED]
 
-### Updated
+### Added
 
-- [#2367](https://github.com/plotly/dash/pull/2367) Updated the default `favicon.ico` to the current Plotly logo
```
Comment (T4rk1n) [inline CHANGELOG.md:9]: The changelog entry for the favicon should be there too.

File: components/dash-html-components/scripts/generate-components.js [hunk 2]
```diff
@@ -263,14 +269,19 @@ const ${Component} = (props) => {
     if(props.loading_state && props.loading_state.is_loading) {
         dataAttributes['data-dash-is-loading'] = true;
     }
+    /* remove unnecessary onClick event listeners  */
+    let isStatic = typeof props.id === "undefined" ? true : false
```
Comment (T4rk1n) [inline components/dash-html-components/scripts/generate-components.js:273]: There is no need for the `? true : false` as it's the condition itself.
Simplified:
```javascript
const isStatic = props.disable_n_clicks || typeof props.id === "undefined";
```

Comment (AnnMarieW) [reply components/dash-html-components/scripts/generate-components.js]: oh yes, that's better.  Thanks!

File: components/dash-html-components/scripts/generate-components.js [hunk 3]
```diff
@@ -263,14 +269,19 @@ const ${Component} = (props) => {
     if(props.loading_state && props.loading_state.is_loading) {
         dataAttributes['data-dash-is-loading'] = true;
     }
+    /* remove unnecessary onClick event listeners  */
+    let isStatic = props.disable_n_clicks || typeof props.id === "undefined";
+    isStatic = typeof props.disable_n_clicks === "undefined" ? isStatic : props.disable_n_clicks
```
Comment (T4rk1n) [inline components/dash-html-components/scripts/generate-components.js:274]: Need to remove the second line.

**Review by T4rk1n — APPROVED** [review]
:woman_dancing:

File: components/dash-html-components/scripts/generate-components.js [hunk 4]
```diff
@@ -263,14 +269,17 @@ const ${Component} = (props) => {
     if(props.loading_state && props.loading_state.is_loading) {
         dataAttributes['data-dash-is-loading'] = true;
     }
-
+    /* remove unnecessary onClick event listeners  */
+    const isStatic = props.disable_n_clicks || typeof props.id === "undefined";
```
Comment (alexcjohnson) [inline components/dash-html-components/scripts/generate-components.js:273]:
```suggestion
    const isStatic = props.disable_n_clicks || !props.id;
```
NBD but we don't allow `id=""` and as a result in dash-renderer we just check if `id` is truthy for this kind of situation.

**Review by alexcjohnson — APPROVED** [review]
Why did the favicon change in this PR? Can you revert that?

After that this looks great! 💃

**Comment by AnnMarieW** [comment]
This pull request was re-created in [2389](https://github.com/plotly/dash/pull/2389) to remove the files included in error.

**Pull request closed without merging**